bridgenode/flatfileworker.go

```diff
@@ -107,47 +119,54 @@ func flatFileWorker(
 	if err != nil {
 		panic(err)
 	}
-	// Grab either proof bytes and write em to offset / proof file, OR, get a TTL result
-	// and write that.  Will this lock up if it keeps doing proofs and ignores ttls?
-	// it should keep both buffers about even.  If it keeps doing proofs and the ttl
-	// buffer fills, then eventually it'll block...?
-	// Also, is it OK to have 2 different workers here?  It probably is, with the
-	// ttl side having read access to the proof writing side's last written proof.
-	// then the TTL side can do concurrent writes.  Also the TTL writes might be
-	// slow since they're all over the place.  Also the offsets should definitely
-	// be in ram since they'll be accessed a lot.
-
-	// TODO ^^^^^^ all that stuff.
-
-	// main selector - Write block proofs whenever you get them
-	// if you get TTLs, write them only if they're not too high
-	// if they are too high, keep writing proof blocks until they're not
-	for {
-		select {
-		case ud := <-proofChan:
-			err = ff.writeProofBlock(ud)
-			if err != nil {
-				panic(err)
-			}
-		case ttlRes := <-ttlResultChan:
-			for ttlRes.Height > ff.currentHeight {
-				ud := <-proofChan
-				err = ff.writeProofBlock(ud)
-				if err != nil {
-					panic(err)
-				}
-			}
 
-			err = ff.writeTTLs(ttlRes)
-			if err != nil {
-				panic(err)
-			}
-		case undo := <-undoChan:
-			err = uf.writeUndoBlock(undo)
-			if err != nil {
-				panic(err)
-			}
-		}
+	undo := <-undoChan
+	err = uf.writeUndoBlock(undo)
+	if err != nil {
+		panic(err)
+	}
+}
+
+func flatFileWorkerTTlBlocks(
+	ttlResultChan chan ttlResultBlock,
+	leafblockChan chan int,
+	utreeDir utreeDir,
+	fileWait *sync.WaitGroup) {
+
+	var tf flatFileState
+	var err error
+
+	tf.offsetFile, err = os.OpenFile(
+		utreeDir.TtlDir.OffsetFile, os.O_CREATE|os.O_RDWR, 0600)
+	if err != nil {
+		panic(err)
+	}
+
+	tf.proofFile, err = os.OpenFile(
+		utreeDir.TtlDir.ttlsetFile, os.O_CREATE|os.O_RDWR, 0600)
+	if err != nil {
+		panic(err)
+	}
+	tf.fileWait = fileWait
+
+	err = tf.ffInit()
+	if err != nil {
+		panic(err)
+	}
+
```

Comment:
We need to add a `for{}` loop for this segment here. Otherwise, the program will do this function exactly once and exit (which is not what we want).

```
for {          
                size := <-leafblockChan
                bytesTtlWrite := make([]byte, size*4)
                _, err = tf.proofFile.WriteAt(bytesTtlWrite, tf.currentOffset)                                          
                tf.fileWait.Done()                                                                                      
                if err != nil {                                                                                         
                        panic(err)
                }         
                                                            
                ttlRes := <-ttlResultChan                   
                err = tf.writeTTLs(ttlRes)               
                if err != nil {
                        panic(err)                       
                }                                                                                                       
                                                            
                tf.currentOffset = tf.currentOffset + int64(size)*4
        }
```

We need something like the code above.